src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -944,14 +919,18 @@ class LicenceService(
     return Pair(firstName, lastName)
   }
 
-  private fun findOriginalLicenceForVariation(variationLicence: EntityLicence): EntityLicence {
+  private fun findOriginalLicenceForVariation(variationLicence: VariationLicence): CrdLicence {
     var originalLicence = variationLicence
     while (originalLicence.variationOfId != null) {
-      originalLicence = licenceRepository
+      val licence = licenceRepository
         .findById(originalLicence.variationOfId!!)
         .orElseThrow { EntityNotFoundException("${originalLicence.variationOfId}") }
+      when (licence) {
+        is CrdLicence -> return licence
+        is VariationLicence -> originalLicence = licence
+      }
     }
-    return originalLicence
+    error("original licence not found for licence: ${variationLicence.id}")
```

Comment:
rewrote this to use information based on the type of logic. This would have been an invalid operation before - a variation chain that started with a variation rather than a CRD licence